### PR TITLE
[KeyVault] Add new parameter `--enable-rbac-authorization` during creating or updating

### PR DESCRIPTION
**Pull request by bim-msft**
Support for #11268 

- [x] Waiting for Swagger/Python SDK, release time: Apr 15.
- [x] Waiting for service public.

How to test:

1. Run `az keyvault create -n {} -g {} --enable-rbac-authorization`
2. Check the value of `properties.enableRbacAuthorization`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/cloud.py [hunk 1]
```diff
@@ -214,6 +214,19 @@ def __str__(self):
...
+    endpoints=CloudEndpoints(
+        management='https://management-preview.core.windows-int.net/',
+        resource_manager='https://api-dogfood.resources.windows-int.net/',
+        gallery='https://df.gallery.azure-test.net/',
+        active_directory='https://login.windows-ppe.net',
+        active_directory_graph_resource_id='https://graph.ppe.windows.net/',
+        active_directory_resource_id='https://management.core.windows.net/',
+        microsoft_graph_resource_id='https://graph.ppe.windows.net/'),
+    suffixes=CloudSuffixes(
+        keyvault_dns='.vault-int.azure-int.net'))
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/cloud.py]: I prefer not to hard-code our testing environment in the public CLI code. We can [register a new cloud](https://docs.microsoft.com/en-us/cli/azure/manage-clouds-azure-cli?toc=%2Fcli%2Fazure%2Ftoc.json&bc=%2Fcli%2Fazure%2Fbreadcrumb%2Ftoc.json&view=azure-cli-latest#register-a-new-cloud) instead.

File: src/azure-cli/azure/cli/command_modules/keyvault/custom.py [hunk 2]
```diff
@@ -453,6 +460,10 @@ def set_policy(cmd, client, resource_group_name, vault_name,
     object_id = _object_id_args_helper(cmd.cli_ctx, object_id, spn, upn)
     vault = client.get(resource_group_name=resource_group_name,
                        vault_name=vault_name)
+
+    if vault.properties.enable_rbac_authorization:
+        raise CLIError('Cannot set policies to a vault with `--enable-rbac-authorization` specified')
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/keyvault/custom.py]: The keyvault may be created via Azure Portal, so customer may not used `--enable-rbac-authorization` to enable RBAC authorization. So how about using 

> Cannot set policies for a vault with RBAC authorization enabled

Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/keyvault/custom.py]: In error message we usually use single quote `'`. Grave accent `` ` `` is only for markdown.

Comment (bim-msft) [reply src/azure-cli/azure/cli/command_modules/keyvault/custom.py]: > The keyvault may be created via Azure Portal, so customer may not used `--enable-rbac-authorization` to enable RBAC authorization. So how about using
> 
> > Cannot set policies for a vault with RBAC authorization enabled

Agree, this is more reasonable, will refine it.

File: src/azure-cli-core/azure/cli/core/cloud.py (hunk 1 above)
Comment (bim-msft) [reply src/azure-cli-core/azure/cli/core/cloud.py]: > I prefer not to hard-code our testing environment in the public CLI code. We can [register a new cloud](https://docs.microsoft.com/en-us/cli/azure/manage-clouds-azure-cli?toc=%2Fcli%2Fazure%2Ftoc.json&bc=%2Fcli%2Fazure%2Fbreadcrumb%2Ftoc.json&view=azure-cli-latest#register-a-new-cloud) instead.

Good idea, I'll deprecate this part.

File: src/azure-cli/azure/cli/command_modules/keyvault/custom.py (hunk 2 above)
Comment (bim-msft) [reply src/azure-cli/azure/cli/command_modules/keyvault/custom.py]: > In error message we usually use single quote `'`. Grave accent `` ` `` is only for markdown.

Good point, will refine this.

**Comment by yonzhan** [comment]
add to S169

File: src/azure-cli/azure/cli/command_modules/keyvault/_help.py [hunk 3]
```diff
@@ -113,7 +113,7 @@
 helps['keyvault create'] = """
 type: command
 short-summary: Create a key vault.
-long-summary: Default permissions are created for the current user or service principal unless the `--no-self-perms` flag is specified.
+long-summary: If `--enable-rbac-authorization` is not specified, then default permissions are created for the current user or service principal unless the `--no-self-perms` flag is specified.
```
Comment (qwordy) [inline src/azure-cli/azure/cli/command_modules/keyvault/_help.py:116]: Long summary is not shown in CLI reference. May you can move it to short summary.

Comment (bim-msft) [reply src/azure-cli/azure/cli/command_modules/keyvault/_help.py:116]: > Long summary is not shown in CLI reference. May you can move it to short summary.

@qwordy Good point, but I think it's ok to keep it as long summary, for the docs display issue, we'd better push docs team to display our long summaries as well. At lease we can see the long summary using `-h`.

Comment (bim-msft) [reply src/azure-cli/azure/cli/command_modules/keyvault/_help.py:116]: image

Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/keyvault/_help.py:116]: OK. No problem.

Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/keyvault/_help.py:116]: This statement is cumbersome.

```suggestion
long-summary: Default permissions are created for the current user or service principal unless the `--no-self-perms` or `--enable-rbac-authorization` flag is specified.
```

**Pull request merged**